IBX-458: Document switching designs (#1411)
ezsystems/developer-documentation: docs/guide/content_rendering/design_engine/add_new_design.md

# Add new design

To create different designs for different version of the website,

> you configure different sites based on the [SiteAccess](../../multisite/multisite.md) concent.

Author: ```suggestion
you configure different sites based on the [SiteAccess](../../multisite/multisite.md) content.
```
Yep, links are broken until the other PR is merged.

This example shows how to prepare a site for a "Summer Sale" marketing campaign
and provide it with a distinct design.

## Configure a new SiteAccess

First, in the SiteAccess configuration in `config/packages/ezplatform.yaml`,
add the `campaign` SiteAccess to the list under `ezplatform.siteaccess`:

``` yaml
ezplatform:
    siteaccess:
        list: [site, campaign]
        groups:
            site_group: [site, campaign]
        default_siteaccess: site
```

> Adding the `campaign` SiteAccess to [`site_group`](../../multisite/multisite_configuration.md#siteaccess-groups) enables you to add common configuration for both SiteAccesses at the same time.

Reviewer: Broken link for Multisite

!!! tip

>     For details about configuring different site roots and matching SiteAccesses, see [Set up campaign SiteAccess](../../multisite/setup_campaign_siteaccess.md).

Reviewer: Broken link

## Create themes

> Next, configure a new `summersale` design this a theme also named `summersale`:

Reviewer: I think some part of this sentence is missing?

Author: ```suggestion
Next, configure a new `summersale` design for this theme, also named `summersale`:
```

``` yaml hl_lines="3 7"
[[= include_file('code_samples/front/add_design/config/packages/views.yaml', 4, 13) =]]
```

Now, create an Article Content item and preview it on the front page.
You should see the article with a header and footer that you defined for the main site.

## Override templates

Now, you need to override the header of the site to fit the campaign.
Create a separate `templates/themes/summersale/parts/header.html.twig` file with different content, for example:

``` html+twig
[[= include_file('code_samples/front/add_design/templates/themes/summersale/parts/header.html.twig') =]]
```

Preview the Article through the `campaign` SiteAccess: `<yourdomain>/campaign/<article-name>`.
You can see that the page uses the campaign header, while the rest of the layout, including the footer,
is the same as in the main site.
This is because you defined `standard` design as fallback for this SiteAccess:

``` yaml
[[= include_file('code_samples/front/add_design/config/packages/views.yaml', 0, 3) =]]
```

In this case, if the design engine cannot find a template for the current design,
it uses the template from the next configured design.

> In the case above, the engine doesn't find the footer template for the `campaign` SiteAccess,

Reviewer: ```suggestion
In the case above, the engine does not find the footer template for the `campaign` SiteAccess,
```
